src/kdtree.rs

```diff
@@ -288,6 +288,25 @@ impl<A: Float + Zero + One, T, U: AsRef<[A]>> KdTree<A, T, U> {
         }
     }
 
+    pub fn remove(&mut self, point: &U, data: &T) -> Result<(), ErrorKind> {
```

Comment:
what's the targeted use case for removing a node?
1) by value(`data`)
2) by coordinates(`point`)
3) OR or BOTH relationship when both value and coordinates are provided

Reply:
I assumed since you need the coordinate and value to create the tree, it would be reasonable to require them both to remove an item. It looked like since the values are stored separately that it would be difficult to remove them without having both.

Reply:
it seems 1 & 2 are also quite common use case
1. e.g. add/remove/update entity's location when `data` represents entity id/uuid
2. e.g. when location(`point`) is actually the primary key for the use case

it's okay if we just want to implement 3's BOTH(AND) relationship but we have to make sure nodes are removed only when both `point` and `data` match the parameters